### PR TITLE
Feature/vg 41 donotlockscreen

### PR DESCRIPTION
**Pull request by KayleeTheMech**
Introduced logarithmic scaling and an unlocked screen position in LIVEVIEW

Large Refactoring of SpectrogramFragment and SpectrogramView

### CODE REVIEW

File: app/src/main/java/de/voicegym/voicegym/menu/settings/SettingsBundle.kt [hunk 1]
```diff
@@ -13,7 +13,7 @@ object SettingsBundle {
     const val sampleRate = 44100
     const val channelConfig = AudioFormat.CHANNEL_IN_MONO
     const val audioFormat = AudioFormat.ENCODING_PCM_16BIT
-    const val normailzationConstant = 55.0
```
Comment (pektinasen) [inline app/src/main/java/de/voicegym/voicegym/menu/settings/SettingsBundle.kt:16]: noticed the typo as well, but it is `normalization`

File: app/src/main/java/de/voicegym/voicegym/recordActivity/RecordActivity.kt [hunk 2]
```diff
@@ -38,6 +41,7 @@ import java.util.Locale
 import java.util.concurrent.ConcurrentLinkedQueue
 import kotlin.concurrent.thread
 
+
```
Comment (pektinasen) [inline app/src/main/java/de/voicegym/voicegym/recordActivity/RecordActivity.kt:44]: this new line is wrong here

File: app/src/main/java/de/voicegym/voicegym/recordActivity/fragments/SpectrogramFragment.kt [hunk 3]
```diff
@@ -23,152 +20,127 @@ class SpectrogramFragment : AbstractInstrumentFragment() {
...
 
-    /**
-     * the interpolator is used to calculate the intensities at the specific pixel positions
-     */
-    private val interpolator = LinearInterpolator()
-
     /**
      * holds the array that stores the frequencies of the relating to the positions in the amplitudeArray
      */
     private var frequencyArray: DoubleArray? = null
```
Comment (pektinasen) [inline app/src/main/java/de/voicegym/voicegym/recordActivity/fragments/SpectrogramFragment.kt:31]: Can you initialize this, so we don't have to deal with null values?

File: app/src/main/java/de/voicegym/voicegym/recordActivity/fragments/SpectrogramFragment.kt [hunk 4]
```diff
@@ -23,152 +20,127 @@ class SpectrogramFragment : AbstractInstrumentFragment() {
...
-
-    /**
-     * this internally stores the range that is displayed from the amplitudeArray
-     */
-    private var internalSpectrogramSettings: RawSpectrogramSettings? = null
+    private var zeroAmplitudes: DoubleArray? = null
```
Comment (pektinasen) [inline app/src/main/java/de/voicegym/voicegym/recordActivity/fragments/SpectrogramFragment.kt:32]: siehe oben

File: app/src/main/java/de/voicegym/voicegym/recordActivity/fragments/SpectrogramFragment.kt [hunk 5]
```diff
@@ -23,152 +20,127 @@ class SpectrogramFragment : AbstractInstrumentFragment() {
...
 
         spectrogramView = view.findViewById(R.id.spectrogramView)
-        spectrogramView.let {
-            it.xDataPoints = settings.displayedDatapoints
-            it.invalidate()
-        }
         return view
     }
 
+
```
Comment (pektinasen) [inline app/src/main/java/de/voicegym/voicegym/recordActivity/fragments/SpectrogramFragment.kt:42]: wrong newline

File: app/src/main/java/de/voicegym/voicegym/recordActivity/views/util/ScalingFunction.kt [hunk 6]
```diff
@@ -0,0 +1,45 @@
+package de.voicegym.voicegym.recordActivity.views.util
+
+import kotlin.math.roundToInt
+
+data class PixelFrequencyPair(val pixelPosition: Int, val correspondingFrequency: Double)
+
+abstract class ScalingFunction(val from: PixelFrequencyPair, val until: PixelFrequencyPair) {
```
Comment (pektinasen) [inline app/src/main/java/de/voicegym/voicegym/recordActivity/views/util/ScalingFunction.kt:7]: why is it an abstract class and not an interface?

File: app/src/main/java/de/voicegym/voicegym/util/audio/PCMPlayer.kt [hunk 7]
```diff
@@ -6,6 +6,7 @@ import android.media.AudioFormat
 import android.media.AudioManager
 import android.media.AudioTrack
 import android.os.Build
+import android.util.Log
```
Comment (pektinasen) [inline app/src/main/java/de/voicegym/voicegym/util/audio/PCMPlayer.kt:9]: I guess this is not needed

**Review by pektinasen — CHANGES_REQUESTED** [review]

File: app/src/main/java/de/voicegym/voicegym/recordActivity/views/util/ScalingFunction.kt (hunk 6 above)
Comment (KayleeTheMech) [reply app/src/main/java/de/voicegym/voicegym/recordActivity/views/util/ScalingFunction.kt:7]: because i intend to store from and until

**Pull request merged**